crits/core/handlers.py

```diff
@@ -1230,6 +1230,69 @@ def toggle_item_state(type_, oid, analyst):
     except ValidationError:
         return {'success': False}
 
+def do_add_preferred_actions(obj_type, obj_id, username):
+    """
+    Add all preferred actions to an object.
+
+    :param obj_type: The type of object to update.
+    :type obj_type: str
+    :param obj_id: The ObjectId of the object to update.
+    :type obj_id: str
+    :param username: The user adding the preferred actions.
+    :type username: str
+    :returns: dict with keys:
+              "success" (boolean),
+              "message" (str) if failed,
+              "object" (list of dicts) if successful.
+    """
+
+    klass = class_from_type(obj_type)
+    if not klass:
+        return {'success': False, 'message': 'Invalid type'}
+
+    preferred_actions = IndicatorAction.objects(preferred=obj_type)
+    if not preferred_actions:
+        return {'success': False, 'message': 'No preferred actions'}
+
+    sources = user_sources(username)
+    obj = klass.objects(id=obj_id, source__name__in=sources).first()
+    if not obj:
+        return {'success': False, 'message': 'Could not find object'}
+
+    actions = []
+    now = datetime.datetime.now()
+    # Get preferred actions and add them.
+    for a in preferred_actions:
+        action = {'action_type': a.name,
+                  'active': 'on',
+                  'analyst': username,
+                  'begin_date': now,
+                  'end_date': now,
```

Comment:
I might just leave the end date as None (or whatever it is that leaves it open-ended)? It would seem odd that an Action would begin and end at the exact same time.